### ISSUE TITLE
GEDI subsetting ancillary fields race condition

### ISSUE THREAD

**Issue by jpswinski** [ICESat2-SlideRule/sliderule#522]
The following issue was reported by a SlideRule user at Texas A&M.

The following code went from taking 5 seconds to 22 minutes when an ancillary field was included:
```Python
parms = {
    "poly": aoi,  
    "t0": '2024-01-01',
    "t1": '2026-01-01',
    "degrade_flag": 0,
    "l2_quality_flag": 1,
    "anc_fields": ["digital_elevation_model"], 
     # "anc_fields": ["rh"],  # Optional: include RH metrics (22min!!!)
 }

L2AP = gedi.gedi02ap(parms)
print(f'Total records returned: {len(L2AP):,}')
L2AP
```

**Comment by jpswinski** [comment]
Upon investigating, there were multiple issues:
* Different number of values were returned with successive calls
* Intermittently, the code would seemingly wait for one of the beams to finish until the request timeout was reached
* The typesize of the ancillary field was sometimes zero, which meant it couldn't be serialized

These odd behaviors were the result of a number of bugs in the code:
* A mutex was not being properly locked on startup
* The ancillary fields map was using the same key for the fields from different beams
* Exceptions in a mutex zone of the code was not being handled properly

These bugs have been fixed in commit d4654933553d6245544c021d27ebbda35c108fab
